src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.ts

```diff
@@ -467,24 +523,19 @@ export const CloudEnvironmentModal = ({
     );
   };
 
-  const renderToolButtons = (toolLabel, cloudProvider) => {
+  const renderToolButtons = (toolLabel: ToolLabel, cloudProvider: CloudProvider) => {
     const app = currentApp(toolLabel);
     const doesCloudEnvForToolExist = currentRuntimeTool === toolLabel || app;
     const isCloudEnvForToolDisabled = isCloudEnvModalDisabled(toolLabel);
     return h(Fragment, [
-      // We cannot attach the periodic cookie setter until we have a running app for Azure because the relay is not guaranteed to be ready until then
-      app?.cloudContext?.cloudProvider === cloudProviderTypes.AZURE && app?.status === 'RUNNING'
-        ? Utils.cond(
-            [toolLabel === appToolLabels.CROMWELL, () => h(PeriodicAzureCookieSetter, { proxyUrl: app.proxyUrls['cbas-ui'], forApp: true })],
-            [
-              toolLabel === appToolLabels.HAIL_BATCH,
-              () => {
-                const batchUrl = app?.proxyUrls?.batch;
-                return h(PeriodicAzureCookieSetter, { proxyUrl: batchUrl.substring(0, batchUrl.lastIndexOf('/') + 1), forApp: true });
-              },
-            ],
-            [Utils.DEFAULT, () => null]
-          )
+      // We cannot attach the periodic cookie setter until we have a running Cromwell app for Azure because the relay is not guaranteed to be ready until then
+      toolLabel === appToolLabels.CROMWELL &&
+      app?.cloudContext?.cloudProvider === cloudProviderTypes.AZURE &&
+      app?.status === 'RUNNING'
+        ? h(PeriodicAzureCookieSetter, {
+            proxyUrl: app.proxyUrls['cbas-ui'],
+            // forCromwell: true //TODO: this is not yet supported behaviour
```

Comment:
Hi -- I think `forApp: true` needs to be passed to `PeriodicAzureCookieSetter` in this case. See previous logic. Also it seems like the functionality to set the cookie for Hail Batch app was lost.

Perhaps the cookie-setting behavior could also be covered in a unit test?